packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanel.tsx

```diff
@@ -544,40 +496,84 @@ export function TransferPanel() {
     }
   }
 
-  const transfer = async () => {
-    const signerUndefinedError = 'Signer is undefined'
-
-    try {
-      setTransferring(true)
-      if (chainId !== networks.sourceChain.id) {
-        await switchNetworkAsync?.(networks.sourceChain.id)
-      }
-    } finally {
-      setTransferring(false)
+  async function hasDestinationAddressError() {
+    const destinationAddressError = await getDestinationAddressError({
+      destinationAddress,
+      isSmartContractWallet,
+      isTeleportMode
+    })
+    if (destinationAddressError) {
+      console.error(destinationAddressError)
+      return true
     }
+    return false
+  }
 
+  async function isTransferAllowed(isCctp: boolean) {
     if (!isConnected) {
-      return
+      return false
     }
     if (!walletAddress) {
-      return
+      return false
+    }
+    if (isWarningToken()) {
+      return false
     }
+    const childChainName = getNetworkName(childChain.id)
+    const isConnectedToTheWrongChain =
+      chainId !== latestNetworks.current.sourceChain.id
 
-    const hasBothSigners = parentSigner && childSigner
-    if (isEOA && !hasBothSigners) {
-      throw signerUndefinedError
+    if (isConnectedToTheWrongChain) {
```

Comment:
I don't know about this entire code block, I'd expect `isTransferAllowed` to be pure. I'd have the network switching logic in its own method.